src/NServiceBus.Core/Transports/Msmq/QueueCreator.cs

```diff
@@ -31,83 +31,57 @@ public Task CreateQueueIfNecessary(QueueBindings queueBindings, string identity)
-            }
-
-            Logger.Debug($"Checking if queue exists: {address}.");
-
-            try
+            MessageQueue queue;
+            if (!MsmqUtilities.TryOpenQueue(msmqAddress, out queue))
             {
-                if (MessageQueue.Exists(queuePath))
-                {
-                    Logger.Debug("Queue exists, going to set permissions.");
-                    SetPermissionsForQueue(queuePath, identity);
-                    return;
-                }
-
                 Logger.Warn("Queue " + queuePath + " does not exist.");
                 Logger.Debug("Going to create queue: " + queuePath);
 
-                CreateQueue(queuePath, identity, settings.UseTransactionalQueues);
-            }
-            catch (MessageQueueException ex)
-            {
-                if (msmqAddress.IsRemote && (ex.MessageQueueErrorCode == MessageQueueErrorCode.IllegalQueuePathName))
+                try
                 {
-                    return;
+                    queue = CreateQueue(queuePath, identity, settings.UseTransactionalQueues);
                 }
-                if (ex.MessageQueueErrorCode == MessageQueueErrorCode.QueueExists)
+                catch (MessageQueueException ex)
                 {
-                    //Solve the race condition problem when multiple endpoints try to create same queue (e.g. error queue).
-                    return;
+                    if (msmqAddress.IsRemote && (ex.MessageQueueErrorCode == MessageQueueErrorCode.IllegalQueuePathName))
+                    {
+                        return;
+                    }
+                    if (ex.MessageQueueErrorCode == MessageQueueErrorCode.QueueExists)
+                    {
+                        //Solve the race condition problem when multiple endpoints try to create same queue (e.g. error queue).
+                        return;
+                    }
+
+                    Logger.Error($"Could not create queue {msmqAddress} or check its existence. Processing will still continue.", ex);
                 }
-                Logger.Error($"Could not create queue {address} or check its existence. Processing will still continue.", ex);
-            }
-            catch (Exception ex)
-            {
-                Logger.Error($"Could not create queue {address} or check its existence. Processing will still continue.", ex);
             }
 
-            Logger.DebugFormat("Verified that the queue: [{0}] existed", address);
-        }
-
-        static void CreateQueue(string queuePath, string account, bool transactional)
-        {
-            using (var queue = MessageQueue.Create(queuePath, transactional))
+            if (queue == null)
             {
-                SetPermissionsForQueue(queue, account);
+                Logger.Error($"Could not create queue {address} or check its existence. Processing will still continue.");
```

Comment:
Also, since you can now tell the difference between failures of the `TryOpenQueue` and `CreateQueue`, the text of the logging message can be more specific.